xdsl/ir.py

```diff
@@ -15,12 +15,41 @@
 OpT = TypeVar('OpT', bound='Operation')
 
 
+@dataclass
+class Dialect:
+    """Contains the operations and attributes of a specific dialect"""
+    __operations: list[Operation] = field(default_factory=list,
+                                          init=True,
+                                          repr=True)
+    __attributes: list[Attribute] = field(default_factory=list,
+                                          init=True,
+                                          repr=True)
+
+    def get_operations(self) -> Iterator[Operation]:
```

Comment:
Should we make this a property instead? I feel it would look better that way.

Reply:
Agree, done.